presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -199,10 +199,14 @@ public static boolean isConstant(Expression expression)
             tempExpression = ((Cast) tempExpression).getExpression();
         }
 
-        if (tempExpression instanceof Literal || tempExpression instanceof ArrayConstructor) {
```

Comment:
The elements in array constructor should also be Literals

Reply:
Can you add a test for this?

Reply:
Added one more test for this in abstracttestqueries

Reply:
This util is used for the reduce_agg function, and it will change the behaviour for the reduce_agg function.
For example, currently the query 
`SELECT id, reduce_agg(value, array[id, value], (a, b) -> a || b, (a, b) -> a || b) FROM ( VALUES (1, 2), (1, 3), (1, 4), (2, 20), (2, 30), (2, 40) ) AS t(id, value) GROUP BY id` will succeed as it considers expression `array[id, value]` as constant.
However, after this fix, it will fail with error `REDUCE_AGG only supports non-NULL literal as the initial value` as `array[id, value]` is not considered as constant now.
cc @kaikalur